Src/AutoFixture/Kernel/FactoryMethodQuery.cs

```diff
@@ -39,6 +43,7 @@ public IEnumerable<IMethod> SelectMethods(Type type)
             return from mi in type.GetMethods(BindingFlags.Static | BindingFlags.Public)
                    where mi.ReturnType == type
                    let parameters = mi.GetParameters()
+                   where mi.GetParameters().All(p => p.ParameterType != type)

```

Comment:
Thank you for the detailed explanation! :+1: 

This basically *improves* the existing heuristic for static Factory Methods, then.